From a220f79133dfb6bfc187f93540f7c1753507f679 Mon Sep 17 00:00:00 2001
From: Viren070 <viren070@protonmail.com>
Date: Mon, 19 Oct 2026 18:19:19 +0100
Subject: [PATCH] avcodec/libdav1d: map skip_loop_filter to dav1d's
 inloop_filters

Discarding the loop filter for all frames turns off dav1d's deblocking,
CDEF and loop restoration, as skip_loop_filter does for the native
decoders.
---
 libavcodec/libdav1d.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libavcodec/libdav1d.c b/libavcodec/libdav1d.c
index 0f713ef..faf68cf 100644
--- a/libavcodec/libdav1d.c
+++ b/libavcodec/libdav1d.c
@@ -240,6 +240,9 @@ static av_cold int libdav1d_init(AVCodecContext *c)
         s.decode_frame_type = DAV1D_DECODEFRAMETYPE_INTRA;
     else if (c->skip_frame >= AVDISCARD_NONREF)
         s.decode_frame_type = DAV1D_DECODEFRAMETYPE_REFERENCE;
+
+    if (c->skip_loop_filter >= AVDISCARD_ALL)
+        s.inloop_filters = DAV1D_INLOOPFILTER_NONE;
 #endif
 
     res = libdav1d_parse_extradata(c);
-- 
2.43.0

